src/registry.rs

```diff
@@ -961,7 +963,7 @@ pub enum CheckKind {
     SysVersionSlice1Referenced,
     // flake8-simplify
     DuplicateIsinstanceCall(String),
-    AAndNotA(String),
+    AAndNotA(flake8_simplify::plugins::AAndNotA),
```

Comment:
I'm feeling dumb but trying to wrap my head around why we still need the `CheckKind` enum at all in this world. I believe we do, but I'm trying to explain why to myself.

Reply:
We could get rid of it if we used boxed trait objects, i.e. `Box<&dyn Violation>` ... method calls on trait objects come with some overhead and boxing requires a heap allocation ... but that overhead is probably negligible. The trait would however need to be changed to be [object safe](https://doc.rust-lang.org/reference/items/traits.html#object-safety) which would mean it could no longer have `Serialize` or `DeserializeOwned` as supertraits because these traits aren't object-safe ... that could be worked around by using something like [erased-serde](https://crates.io/crates/erased-serde).

As long as we don't need to use trait objects I think continuing to use an enum is easier.

Reply:
I agree.